src/Storages/StorageURL.cpp

```diff
@@ -1327,6 +1328,7 @@ std::optional<time_t> IStorageURLBase::tryGetLastModificationTime(
                    .withBufSize(settings.max_read_buffer_size)
                    .withRedirects(settings.max_http_get_redirects)
                    .withHeaders(headers)
+                   .withProxy(proxy_config)
```

Comment:
cc @CheSema I believe you forgot to include proxy configuration https://github.com/ClickHouse/ClickHouse/pull/58845